### DISCUSSION TITLE
How to debug watcher (`--watch` mode)?

### DISCUSSION THREAD

**Discussion by jerrygreen** [avajs/ava#2707] [Q&A] [answered]
When I use `--watch` mode, in some cases I'm getting multiple test reruns when I expect only one rerun. It adds a lot of confusion and too much output into console, but I can't figure why it's happening.

I followed [the docs](https://github.com/avajs/ava/blob/main/docs/recipes/watch-mode.md#debugging) and tried `--verbose` option to analyze that, - I thought it would show reason why any particular run was called (i.e. what particular file change initiated the watcher), but it doesn't show why watcher was initiated.

I would gladly add those to my `ignoredByWatcher` config, but I just can't figure out what to add there.

**Comment by novemberborn** [comment, marked as answer]
It's the `DEBUG=ava:watcher` bit that you should need. Let me know if that works for you.

Looking at those docs we should move away from the `npm test` stuff to `npx ava` and maybe even remove the `--verbose` bit. Perhaps I could interest you in opening a PR for that? 😄

**Reply by jerrygreen** [reply]
I thought I tried `DEBUG=ava:watcher` but somehow I now figured it indeed works! Maybe it was too late last time I tried, and my mind was making tricks on me 😂 I now understand that the second time it's initiated by some compiled `.wasm` module, and the first one is probably just initial run. Thanks!
